src/challenge/Challenge.tsx

```diff
@@ -56,16 +58,24 @@ class Challenge extends React.Component<ChallengeProps, ChallengeState> {
   editorRef = React.createRef<PyEditor>();
 
   currentConsoleText: string = "";
+  currentGraphicsCommand: string = "";
+
   printCallback = throttle(
     () => this.setState({ consoleText: this.currentConsoleText }),
     100
   );
 
+  drawCallback = throttle(
+    () => this.setState({ graphicsCommand: this.currentGraphicsCommand }),
+    100
+  );  
+
```

Comment:
I don't think this should be communicating the command via setState (which triggers a React redraw). Using throttle will prevent performance issues, but you might lose some calls (I think that's what is happening with the arc drawing).

Consider just setting up a reference to the Graphics pane (see how it's done for PyEditor), and call a method on it.